### ISSUE TITLE
Flake8 does not report F841 violations when variables are unpacked from a function return

### ISSUE THREAD

**Issue by Mahdi-Hosseinali** [PyCQA/flake8#1807]
### how did you install flake8?

```console
pip install flake8
```

### unmodified output of `flake8 --bug-report`
On my local machine:
```json
{
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.8.13",
    "system": "Darwin"
  },
  "plugins": [
    {
      "plugin": "mccabe",
      "version": "0.7.0"
    },
    {
      "plugin": "pycodestyle",
      "version": "2.10.0"
    },
    {
      "plugin": "pyflakes",
      "version": "3.0.1"
    }
  ],
  "version": "6.0.0"
}
```
On an AWS EC2 instance:
```json
{
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.8.10",
    "system": "Linux"
  },
  "plugins": [
    {
      "plugin": "mccabe",
      "version": "0.7.0"
    },
    {
      "plugin": "pycodestyle",
      "version": "2.10.0"
    },
    {
      "plugin": "pyflakes",
      "version": "3.0.1"
    }
  ],
  "version": "6.0.0"
}
```

#### what I expected to happen
Get a F841 violation for all variables that are assigned but are unused, including variable unpacking of a function. Pylance correctly picks up these violations.

#### sample code

```python
def single_func():
    return 1


def multiple_func():
    return 1, 2


def main():
    ans1 = single_func()

    ans1_used = single_func()
    print(ans1_used)

    ans1_1_unused, ans1_2_unused = multiple_func()

    ans1_1_unused, ans1_2_used = multiple_func()
    print(ans1_2_used)

    ans1_1_used, ans1_2_used = multiple_func()
    print(ans1_1_used, ans1_2_used)

    unpack1, unpack2 = (3, 4)


if __name__ == "__main__":
    main()

```

#### commands ran
`flake8 t.py`

```console
$ flake8 t.py
t.py:10:5: F841 local variable 'ans1' is assigned to but never used
t.py:23:5: F841 local variable 'unpack1' is assigned to but never used
t.py:23:14: F841 local variable 'unpack2' is assigned to but never used
```

**Comment by asottile** [comment]
please read the prompts on the issue tracker and the bold text at the top of the issue template
